app/src/main/java/com/github/sdp/mediato/CreateProfileFragment.java

```diff
@@ -35,154 +33,160 @@
+  }
+
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup container,
+      Bundle savedInstanceState) {
+    View view = inflater.inflate(R.layout.fragment_create_profile, container, false);
+
+    final TextInputLayout usernameTextInput = view.findViewById(R.id.username_text_input);
+    final TextInputEditText usernameEditText = view.findViewById(R.id.username_edit_text);
+    final MaterialButton createProfileButton = view.findViewById(R.id.create_profile_button);
+    final FloatingActionButton profileImageButton = view.findViewById(
+        R.id.profile_image_add_button);
+
+    profileImage = view.findViewById(R.id.profile_image);
+    photoPicker = new PhotoPicker(this, profileImage);
+
+    // Open a photo picker to choose the profile image
+    profileImageButton.setOnClickListener(v ->
+        photoPicker.getOnClickListener(requireActivity().getActivityResultRegistry()).onClick(v)
+    );
+
+    // Generate a username
+    usernameTextInput.setEndIconOnClickListener(
+        generateUsername(usernameTextInput, usernameEditText));
+
+    // Remove the error if the user fix its username
+    usernameEditText.addTextChangedListener(
+        new UsernameWatcher(usernameTextInput, usernameEditText));
+
+    // Create the profile if valid otherwise error
+    createProfileButton.setOnClickListener(tryCreateProfile(usernameTextInput, usernameEditText));
+
+    // Inflate the layout for this fragment
+    return view;
+  }
+
+  @NonNull
+  private View.OnClickListener generateUsername(TextInputLayout usernameTextInput,
+      TextInputEditText usernameEditText) {
+    return v -> {
+      usernameTextInput.setError(null);
+      Faker faker = new Faker();
+      String animal = faker.animal().name();
+      String number = faker.number().digits(5);
+      usernameEditText.setText(animal.concat(number));
+    };
+  }
+
+  @NonNull
+  private View.OnClickListener tryCreateProfile(TextInputLayout usernameTextInput,
+      TextInputEditText usernameEditText) {
+    return view -> {
+      String errorMsg = getUsernameErrorMsg(usernameEditText.getText());
+      usernameTextInput.setError(errorMsg);
+      if (null == errorMsg) {
+        //TODO Add navigation to Add favorite page
+        //Creating new user and adding it to the database
+        String username = Objects.requireNonNull(usernameEditText.getText()).toString();
+        userBuilder.setUsername(username);
+        userBuilder.setRegisterDate(Dates.getToday());
+        //@TODO by default the location is not set - to be changed when we implement the GPS feature
+        userBuilder.setLocation(new Location());
+        User user = userBuilder.build();
+        Uri profilePicUri = photoPicker.getProfileImageUri();
+        Database.addUser(user);
+        if (photoPicker.getProfileImageUri() != null) {
+          uploadProfilePicTask = Database.setProfilePic(user.getUsername(), profilePicUri);
+        }
+        try {
+          Thread.sleep(500);
```

Comment:
For this placeholder delay, it would be nice to double it because we saw it was non-effectice. I'll push the changes since we already discussed it !